lib/CAPI/TTAttrs.cpp

```diff
@@ -107,7 +109,13 @@ MlirAttribute ttmlirTTSystemDescAttrGet(
         mlir::cast<ChipChannelAttr>(unwrap(chipChannel)));
   }
 
-  return wrap(SystemDescAttr::get(unwrap(ctx), chipDescsUnwrapped,
+  std::vector<tt::CPUDescAttr> cpuDescsUnwrapped;
+  for (auto cpuDesc : cpuDescsRef)
+  {
+    cpuDescsUnwrapped.push_back(mlir::cast<CPUDescAttr>(unwrap(cpuDesc)));
```

Comment:
this function is another case where IMO all of these should be `emplace_back` not `push_back` (though here I conformed to existing style), fwiw

Reply:
My use of `push_back` is mostly just a force of habit. I agree `emplace_back` is more efficient, unsure if we want to escalate this for the rest of the codebase. I'd ask others for their opinions first.

Reply:
Feel free to use `emplace_back`, I think a lot of this serialization code isn't in the hot path, but if there's an opportunity to be more efficient without a hit to readability go for it! :)